### PR TITLE
[extension/auth] split into its own module

### PR DESCRIPTION
**Pull request by codeboten**
Linked issue #7054

### CODE REVIEW

File: extension/ballastextension/go.mod [hunk 1]
```diff
@@ -53,6 +53,8 @@ replace go.opentelemetry.io/collector/exporter => ../../exporter
 
 replace go.opentelemetry.io/collector/extension => ../
 
+replace go.opentelemetry.io/collector/extension/auth => ../auth
```
Comment (jpkrohling) [inline extension/ballastextension/go.mod:56]: Interesting, why does the ballast extension has a dependency on the auth package?

**Review by jpkrohling — APPROVED** [review]

File: extension/ballastextension/go.mod (hunk 1 above)
Comment (codeboten) [reply extension/ballastextension/go.mod:56]: it does not... crosslink adds the replace in here automatically

Comment (bryan-aguilar) [reply extension/ballastextension/go.mod:56]: Is there any chance that this is a transient dependency?

Comment (jpkrohling) [reply extension/ballastextension/go.mod:56]: I removed the replace statement, ran the tests, and it worked without implying further changes to go.sum nor go.mod. Meaning: it's not a transitive dependency.

**Pull request merged**